### ISSUE TITLE
MIDI not working on Windows

### ISSUE THREAD

**Issue by fand** [fand/veda#33]
reported by la_gauche, thx!

<details>
<summary>Talk log on livecode.slack.com</summary>

```
la_gauche [8:19 AM] 
@U6C6GQYG1 do I need to do anything in particular to get your midi.frag or note.frag example working? I'm sending midi out of tidal using loopMIDI. It's a windows virtual midi router if you aren't familiar. Chrome is picking it up. I've tested a few random synths in webpages. I'm sending out notes, but nothing is happening in my Atom background. Just a black screen. Not sure if I have to install a plugin on top of Veda so that it picks up Web Midi (edited)
1 reply Today at 10:07 AM View thread


amagitakayosi [10:07 AM] 
It doesn't need any plugins. Maybe it's a problem of MIDI status number.
open this page and try MIDI events then show me the logs, plz :pray: https://mikatahara.github.io/SendReceive/


la_gauche [10:07 AM] 
ok ill try now


la_gauche [10:10 AM] 
uploaded and commented on this image: web midi.png
1 Comment
@U6C6GQYG1 I had to change to a different port at the top, but otherwise it's receiving messages. I don't know what CC or notes I should send note.frag or midi.frag if that matters. I'm just sending many different notes with different velocities for my test.



la_gauche [10:21 AM] 
I don't know GLSL obviously : )


amagitakayosi [10:24 AM] 
hm.. it seems nothing's wrong.. virtual MIDI keys on my mac shows almost same logs but note.frag works. Is there any errors when you open Devtools on Atom?


la_gauche [10:25 AM] 
I don't think so, but I haven't used dev tools. Other *.frag examples work great and many from the glsl sandbox website


[10:25] 
I'm on Windows 10 64bit if that matters


amagitakayosi [10:28 AM] 
I said `nothing's wrong` to the screenshot :sweat_smile:


la_gauche [10:30 AM] 
uploaded and commented on this image: dev tools.png
1 Comment
A pic of note.frag and dev tools open



la_gauche [10:31 AM] 
i think dev tools says there is one html error, but I don't think that's related. (edited)


amagitakayosi [10:36 AM] 
ah you're trying tidal-midi! cool


[10:37] 
The MIDI port must be initialized before launching Veda, plz try reloading Atom, init Tidal, make sure the MIDI port is ready, then `veda:toggle`.


[10:38] 
if it doesn't work it must be a bug. I'll open an issue.


new messages
la_gauche [10:47 AM] 
@U6C6GQYG1 oh yes I made sure the midi port is initialized first then I toggle Veda, but no luck. Maybe it is a bug indeed. It could be a Windows bug also : ( I've seen Tidal-Midi + Veda working on a mac
```
</details>

**Comment by fand** [comment]
I tried win10 on VirtualBox and found a weird behavior.
When I open Atom it MIDI doesn't work, but after reloading the window it works.

![virtualbox_msedge - win10_07_11_2017_16_50_48](https://user-images.githubusercontent.com/1403842/32483493-a3bf8b20-c3df-11e7-8d1b-5933e59605fe.png)

This is the setting of loopMIDI and virtual MIDI keyboard.
![virtualbox_msedge - win10_07_11_2017_16_51_20](https://user-images.githubusercontent.com/1403842/32483497-a70f7e3e-c3df-11e7-8744-9e7f45a3181b.png)

**Comment by fand** [comment]
when I registered `input.onmidimessage` after `onstatechange` it worked!
I'll fix it and release soon

**Comment by fand** [comment]
fixed in v1.1.1
